extension/src/popup/views/AddAccount/AddAccount/index.tsx

```diff
@@ -35,30 +36,60 @@ export const AddAccount = () => {
   const { t } = useTranslation();
   const dispatch: AppDispatch = useDispatch();
   const authError = useSelector(authErrorSelector);
+  const hasPrivateKey = useSelector(hasPrivateKeySelector);
 
-  const handleSubmit = async (values: FormValues) => {
-    const { password } = values;
+  // In case a password is not provided here popupMessageListener/addAccount
+  // will try to use the existing password value saved in the session store
+  const handleAddAccount = useCallback(
+    async (password: string = "") => {
+      const res = await dispatch(addAccount(password));
 
-    const res = await dispatch(addAccount(password));
+      if (addAccount.fulfilled.match(res)) {
+        emitMetric(METRIC_NAMES.accountScreenAddAccount, {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          number_of_accounts: res.payload.allAccounts.length,
+        });
+        navigateTo(ROUTES.account);
+      }
+    },
+    [dispatch],
+  );
 
-    if (addAccount.fulfilled.match(res)) {
-      emitMetric(METRIC_NAMES.accountScreenAddAccount, {
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        number_of_accounts: res.payload.allAccounts.length,
-      });
-      navigateTo(ROUTES.account);
-    }
+  const handleEnterPassword = async (values: FormValues) => {
+    await handleAddAccount(values.password);
   };
 
+  // If we have a private key we can assume the user password is also saved in
+  // the current session store, so no need to ask for it again
+  useEffect(() => {
+    if (hasPrivateKey) {
+      handleAddAccount();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   useEffect(
     () => () => dispatch(clearApiError()) as unknown as void,
     [dispatch],
   );
 
+  // No need to ask for password if it's already stored, so let's just briefly
+  // display a loading screen while user is being redirected to next screen
+  if (hasPrivateKey) {
+    return (
+      <React.Fragment>
+        <SubviewHeader title="" />
+        {!authError && <Loading />}
```

Comment:
I'm not sure if this is worth a change or not but just for thought -

We have other places in Freighter that use a similar pattern but also produce the "loading stutter" effect we get when creating an account here. This happens because there are 2 loaders for 1 transition in state.

Is there an easy way to make that feel more like 1 loading state?

Reply:
@aristidesstaffieri Hummm I think I know what you mean. I _believe_ there is probably a way of making this transition look nicer but I'm _not sure_ it'll be a straightforward fix.

I'm still getting used to the codebase and to the React Web syntax (which is different from React Native syntax) so I think it'd take some time for me to figure this out. In case you think it's not a blocker, I'd like to move ahead "as is" and create a ticket so we could tackle it later. What do you think?

Reply:
Thanks for the review btw!

Reply:
sure thing and yeah I think it's not a blocker, I do wonder if it's better to leave this loading state off for now. It feels a bit less choppy to me and I don't see that we had one in this flow before.

Reply:
@aristidesstaffieri  on [this commit](https://github.com/stellar/freighter/pull/1685/commits/2df35014371a5ef29b43aecc2d4d7041fb05e4df) I've removed the duplicate spinner so the UX is looking like the below. I think it's looking less chopy now :)

https://github.com/user-attachments/assets/f1ba36ce-aa01-432d-a499-8256a4887265

Reply:
nice, thanks for entertaining that. We can think about how to improve the lag between the transition in a follow up.